osbs/build/user_params.py

```diff
@@ -0,0 +1,207 @@
+            logger.warning("build_image or build_imagestream is defined, they are deprecated,"
+                           "use build_from instead")
+
+        if build_from:
+            source_type, source_value = build_from.split(':', 1)
+            if source_type not in ('image', 'imagestream'):
+                raise OsbsValidationException(
+                    'first part in build_from, may be only image or imagestream')
+            if source_type == 'image':
+                self.build_image.value = source_value
+            else:
+                self.build_imagestream.value = source_value
+
+        self.platforms.value = platforms
+        self.platform.value = platform
+        self.koji_target.value = koji_target
+        self.koji_task_id.value = koji_task_id
+        self.filesystem_koji_task_id.value = filesystem_koji_task_id
+        self.koji_parent_build.value = koji_parent_build
+        self.flatpak.value = flatpak
+        self.flatpak_base_image.value = flatpak_base_image
+        self.isolated.value = isolated
+        self.scratch.value = scratch
+
+        if not flatpak:
+            if not base_image:
+                raise OsbsValidationException("base_image must be provided")
+            self.trigger_imagestreamtag.value = get_imagestreamtag_from_image(base_image)
+
+            if not name_label:
+                raise OsbsValidationException("name_label must be provided")
+            self.imagestream_name.value = name_label.replace('/', '-')
+
+        if signing_intent and compose_ids:
+            raise OsbsValidationException(
+                'Please only define signing_intent -OR- compose_ids, not both')
+        if compose_ids and yum_repourls:
+            raise OsbsValidationException(
+                'Please only define yum_repourls -OR- compose_ids, not both')
+        if not (compose_ids is None or isinstance(compose_ids, list)):
+            raise OsbsValidationException("compose_ids must be a list")
+        if not (yum_repourls is None or isinstance(yum_repourls, list)):
+            raise OsbsValidationException("yum_repourls must be a list")
+        self.yum_repourls.value = yum_repourls or []
+        self.signing_intent.value = signing_intent
+        self.compose_ids.value = compose_ids or []
+
+        if additional_tag_data:
+            self.additional_tags = AdditionalTagsConfig(additional_tag_data['dir_path'],
+                                                        additional_tag_data['file_name'],
+                                                        additional_tag_data['tags'])
+        self._populate_image_tag()
+
+    def __repr__(self):
+        return "UserParams(%s)" % self.__dict__
+
+    def from_json(self, user_params_json):
+        if not user_params_json:
+            return
+        json_dict = json.loads(user_params_json)
+        logger.debug("CALLED FROM JSON HERE with dict %s", json_dict)
+        for key, value in json_dict.items():
```

Comment:
Can we just call `self.set_params(**json_dict)` ?

Reply:
base_image isn't in the json, so that would crash non-flatpak builds.

Reply:
It might be easier to pluck out parameters that need special handling rather than maintaining a list of everything that needs to be serialized though.

Reply:
I've used your suggestion below to create convert_dict during __init__ from the BuildParams, so we don't have to maintain the list.